### PR TITLE
[Concept Entry] PyTorch: Built-in Loss Functions

### LINKED ISSUE #5861: [Concept Entry] PyTorch: Built-in Loss Functions
### Reason/inspiration (optional)

We would like a new entry on `Built-in Loss Functions` concept under PyTorch. The entry should go in a new file under `docs/content/pytorch/concepts/built-in-loss-functions/built-in-loss-functions.md`.       
  
The entry should include:      
  
- An introduction to the concept      
- A `Syntax` section that provides the syntax for the concept (if applicable)      
- An `Example` section that provides an example demonstrating the concept in use (if applicable)     

 
Please refer to the [concept entry template](https://github.com/Codecademy/docs/blob/main/documentation/concept-entry-template.md), [content standards](https://github.com/Codecademy/docs/blob/main/documentation/content-standards.md), and [markdown style guide](https://curriculum-documentation.codecademy.com/content-guidelines/markdown-style-guide/) when working on the PR for this issue.|

### Entry Type (select all that apply)

New Concept Entry for Existing Topic

### Code of Conduct

- [X] By submitting this issue, I agree to follow Codecademy Doc's [Code of Conduct](https://github.com/Codecademy/docs/blob/main/.github/CODE_OF_CONDUCT.md).

### For Maintainers

- [X] Labels added
- [ ] Issue is assigned

### PR DESCRIPTION
**Pull request by PragatiVerma18**
### Description

Add a new entry on Built-in Loss Functions concept under PyTorch.

### Issue Solved

Closes #5861

### Type of Change

- [x] Adding a new entry
- [ ] Editing an existing entry (fixing a typo, bug, issues, etc)
- [ ] Updating the documentation

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] I have linked any issues that are relevant to this PR in the `Issues Solved` section.

### CODE REVIEW

File: content/pytorch/concepts/built-in-loss-functions/built-in-loss-functions.md [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+The syntax is as follows:
+
+```pseudo
+torch.nn.CrossEntropyLoss(weight=None, size_average=None, ignore_index=-100, reduce=None, reduction='mean')
+```
+
+- `weight` (Tensor, optional): A manual rescaling weight given to each class. It has to be a tensor of size `C` where `C` is the number of classes. Default is `None`, meaning no rescaling.
+- `size_average` (bool, optional): Deprecated (use `reduction` instead). If `True`, the loss is averaged over observations for each mini-batch. Default is `True`.
+- `ignore_index` (int, optional): Specifies a target value that is ignored and does not contribute to the loss calculation.
+- `reduce` (bool, optional): Deprecated (use `reduction` instead). If `True`, it sums the losses across the batch. Default is `True`.
```
Comment (avdhoottt) [inline content/pytorch/concepts/built-in-loss-functions/built-in-loss-functions.md:63]: If these are deprecated, then I think we don't need to include them.

File: content/pytorch/concepts/built-in-loss-functions/built-in-loss-functions.md [hunk 2]
```diff
@@ -0,0 +1,219 @@
...
+
+```py
+import torch
+import torch.nn as nn
+
+loss_fn = nn.MSELoss()
+predictions = torch.tensor([2.0, 3.0, 4.0])
+targets = torch.tensor([2.5, 3.5, 4.5])
+loss = loss_fn(predictions, targets)
+print(loss)  # Output: tensor(0.0833)
```
Comment (avdhoottt) [inline content/pytorch/concepts/built-in-loss-functions/built-in-loss-functions.md:47]: Recheck the output of this code here. I'm getting this output - `tensor(0.2500)`

File: content/pytorch/concepts/built-in-loss-functions/built-in-loss-functions.md [hunk 3]
```diff
@@ -0,0 +1,219 @@
...
+
+```py
+import torch
+import torch.nn as nn
+
+loss_fn = nn.CrossEntropyLoss()
+logits = torch.tensor([[1.0, 2.0, 3.0], [1.0, 2.0, 3.0]])
+labels = torch.tensor([2, 0])
+loss = loss_fn(logits, labels)
+print(loss)  # Output: tensor(0.4076)
```
Comment (avdhoottt) [inline content/pytorch/concepts/built-in-loss-functions/built-in-loss-functions.md:79]: I'm getting this output for the above code - `tensor(1.4076)`

File: content/pytorch/concepts/built-in-loss-functions/built-in-loss-functions.md [hunk 4]
```diff
@@ -0,0 +1,219 @@
...
+```py
+import torch
+import torch.nn as nn
+
+# Example of BCEWithLogitsLoss
+loss_fn = nn.BCEWithLogitsLoss()
+logits = torch.tensor([0.5, -1.5, 2.0])
+labels = torch.tensor([1.0, 0.0, 1.0])
+loss = loss_fn(logits, labels)
+print(loss)  # Output: tensor(0.4891)
```
Comment (avdhoottt) [inline content/pytorch/concepts/built-in-loss-functions/built-in-loss-functions.md:111]: Recheck the output.

File: content/pytorch/concepts/built-in-loss-functions/built-in-loss-functions.md [hunk 5]
```diff
@@ -0,0 +1,219 @@
...
+import torch
+import torch.nn as nn
+
+# Example of CosineEmbeddingLoss
+loss_fn = nn.CosineEmbeddingLoss()
+input1 = torch.tensor([1.0, 0.0])
+input2 = torch.tensor([0.0, 1.0])
+target = torch.tensor([1])  # 1 means the inputs should be similar
+loss = loss_fn(input1, input2, target)
+print(loss)  # Output: tensor(2.0)
```
Comment (avdhoottt) [inline content/pytorch/concepts/built-in-loss-functions/built-in-loss-functions.md:175]: It's giving a runtime error. Please correct the code.

File: content/pytorch/concepts/built-in-loss-functions/built-in-loss-functions.md [hunk 6]
```diff
@@ -0,0 +1,219 @@
...
+import torch
+import torch.nn as nn
+
+# Example of KLDivLoss
+loss_fn = nn.KLDivLoss(reduction='batchmean')
+input = torch.tensor([[0.0, 0.1, 0.2], [0.1, 0.0, 0.3]])
+target = torch.tensor([[0.0, 0.1, 0.3], [0.1, 0.2, 0.2]])
+loss = loss_fn(input.log(), target)
+print(loss)  # Output: tensor(0.0237)
+```
```
Comment (avdhoottt) [inline content/pytorch/concepts/built-in-loss-functions/built-in-loss-functions.md:207]: Something wrong with the code snippet. It's giving `tensor(nan)` output.

File: content/pytorch/concepts/built-in-loss-functions/built-in-loss-functions.md [hunk 7]
```diff
@@ -0,0 +1,219 @@
...
+
+## Choosing the Right Loss Function
+
+When selecting a loss function for your model, it is essential to consider the task you're working on.
```
Comment (avdhoottt) [inline content/pytorch/concepts/built-in-loss-functions/built-in-loss-functions.md:211]: Please don't use first and second-person pronouns.

**Review by avdhoottt — CHANGES_REQUESTED** [review]
Hey @PragatiVerma18, I've reviewed the entry and left some comments. Please make the changes, thank you!!

**Comment by PragatiVerma18** [comment]
Hey @avdhoottt, all changes are done. Please review. Thanks.

**Review by avdhoottt — APPROVED** [review]
@PragatiVerma18, Thank you!! LGTM!!
